examples/text-classification/new_tf_text_classification.py

```diff
@@ -0,0 +1,498 @@
+#!/usr/bin/env python
+# coding=utf-8
+# Copyright 2020 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Fine-tuning the library models for sequence classification."""
+# You can also adapt this script on your own text classification task. Pointers for this are left as comments.
+
+import logging
+import os
+import random
+import sys
+from dataclasses import dataclass, field
+from typing import Optional
+from math import ceil
+from pathlib import Path
+
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '1'  # Reduce the amount of console output from TF
+
+import numpy as np
+import tensorflow as tf
+from datasets import load_dataset, load_metric
+
+import transformers
+from transformers import (
+    AutoConfig,
+    TFAutoModelForSequenceClassification,
+    AutoTokenizer,
+    DataCollatorWithPadding,
+    EvalPrediction,
+    HfArgumentParser,
+    PretrainedConfig,
+    TrainingArguments,
+    default_data_collator,
+    set_seed,
+)
+from transformers.trainer_utils import is_main_process, get_last_checkpoint
+from transformers.utils import check_min_version
+
+logger = logging.getLogger(__name__)
+
+# region Helper classes
+class DataSequence(tf.keras.utils.Sequence):
+    # We use a Sequence object to load the data. Although it's completely possible to load your data as Numpy/TF arrays
+    # and pass those straight to the Model, this constrains you in a couple of ways. Most notably, it requires all
+    # the data to be padded to the length of the longest input example, and it also requires the whole dataset to be
+    # loaded into memory. If these aren't major problems for you, you can skip the sequence object in your own code!
+    def __init__(self, dataset, non_label_column_names, batch_size, labels, shuffle=True):
+        super().__init__()
+        # Retain all of the columns not present in the original data - these are the ones added by the tokenizer
+        self.data = {key: dataset[key] for key in dataset.features.keys()
+                     if key not in non_label_column_names
+                     and key != 'label'}
+        data_lengths = {len(array) for array in self.data.values()}
+        assert len(data_lengths) == 1  # Assert that all arrays have the same length
```

Comment:
As a general style observation, we prefer not to have empty asserts in the code base (applies below as well)
```suggestion
        assert len(data_lengths) == 1, "Error message"  # Assert that all arrays have the same length
```